### PR TITLE
[Fix] Reformat and fix Concept/Resource example code

### PR DESCRIPTION
**Pull request by yukicode**
Reformatted example code to meet documentation guideline. 
Fixed a wrong return type in comment.
Made the example code compile.

### CODE REVIEW

File: book/src/concepts/resource.md [hunk 1]
```diff
@@ -61,10 +71,20 @@ If you want to change a resource that is already inside of `Resources`:
...
-let mut fetched = resources.try_fetch_mut::<MyResource>().expect("No MyResource present in Resources");
+#   let my = MyResource{
+#     game_score: 0,
+#   };
+#   resources.insert(my);
+  // try_fetch_mut returns a Option<FetchMut<MyResource>>
+  let mut fetched = resources.try_fetch_mut::<MyResource>()
+                             .expect("No MyResource present in Resources");
+  assert_eq!(fetched.game_score, 0);
+  fetched.game_score = 10;
```
Comment (jojolepro) [inline book/src/concepts/resource.md]: Could you add another assert_eq after this line showing that the score is now 10?
Thanks!

Comment (yukicode) [reply book/src/concepts/resource.md]: Sure! I was debating between adding it or not :)

File: book/src/concepts/resource.md [hunk 2]
```diff
@@ -36,11 +36,20 @@ Fetching a resource can be done like this:
...
 #   let mut resources = Resources::new();
-// Returns a Option<MyResource>
-let fetched = resources.try_fetch::<MyResource>().expect("No MyResource present in Resources");
+#   let my = MyResource{
+#     game_score: 0,
+#   };
+#   resources.insert(my);
+  // try_fetch returns a Option<Fetch<MyResource>>
+  let fetched = resources.try_fetch::<MyResource>()
+                       .expect("No MyResource present in Resources");
```
Comment (torkleyy) [inline book/src/concepts/resource.md]: Would be nice to not panic here, because that's the point of `try_fetch` ;)

Comment (yukicode) [reply book/src/concepts/resource.md]: Good point. I could use 
`If let Some(mut fetched_resource)... else...` 
to replace the expect().
What do you think?

Comment (torkleyy) [reply book/src/concepts/resource.md]: Sounds good!

Comment (yukicode) [reply book/src/concepts/resource.md]: Changed :)

**Pull request merged**